EDQueue/EDQueueStorageEngine.m

```diff
@@ -52,14 +52,32 @@ - (void)dealloc
  *
  * @return {void}
  */
-- (void)createJob:(id)data forTask:(id)task
+- (void)createJob:(id)data forTask:(id)task error:(NSError * __autoreleasing *)outError
 {
     NSString *dataString = [[NSString alloc] initWithData:[NSJSONSerialization dataWithJSONObject:data options:NSJSONWritingPrettyPrinted error:nil] encoding:NSUTF8StringEncoding];
-    
+
+    __block BOOL dbError = false;
+    __block int lastErrorCode = 0;
+    __block NSString *lastErrorMessage = @"";
     [self.queue inDatabase:^(FMDatabase *db) {
         [db executeUpdate:@"INSERT INTO queue (task, data) VALUES (?, ?)", task, dataString];
-        [self _databaseHadError:[db hadError] fromDatabase:db];
+
+        dbError = [db hadError];
+        if (dbError) {
+            lastErrorCode = [db lastErrorCode];
+            lastErrorMessage = [db lastErrorMessage];
+        }
+
+        [self _databaseHadError:dbError fromDatabase:db];
     }];
+
+    if (dbError && outError) {
```

Comment:
Will this ever be truthful? Won't `outError` be a nil reference here? I might be misremembering Obj-C stuff.

Reply:
No out error will be an address to a nil

Reply:
It's only false if the user explicitly passes in a `nil` for the outError. Because outError is the address of the pointer to a space in memory where NSError should be.

Reply:
Radical